routes/auth.routes.js

```diff
@@ -0,0 +1,83 @@
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const mongoose = require("mongoose");
+const User = require("../models/User.model");
+const { isLoggedIn, isLoggedOut } = require("../middleware/route-guard");
+
+router.get("/signup", (req, res) => res.render("auth/signup"));
+
+router.post("/signup", async (req, res, next) => {
+  try {
+    let { username, password } = req.body;
+
+    if (!username || !password) {
+      res.render("auth/signup", { errorMessage: "Fill out all the fields" });
+    }
+
+    const salt = await bcrypt.genSalt(10);
+
+    const hashedPassword = await bcrypt.hash(password, salt);
+
+    await User.create({ username, password: hashedPassword });
+
+    res.redirect("/");
+  } catch (error) {
+    if (error instanceof mongoose.Error.ValidationError) {
+      res.render("auth/signup", {
+        errorMessage: error.message,
+      });
+
+      console.log(error);
+      next(error);
+    }
+  }
+});
+
+// Se incluir o isLoggedOut a página de login não abre, não consegui resolver este problema.
```

Comment:
It breaks if you had the `isLoggedOut` because you were missing the `next()`, in your 'route-guard' file, so, your website would stay stuck in the middleware.